src/test/scala/cromwell/webservice/CromwellApiServiceSpec.scala

```diff
@@ -101,8 +109,46 @@ class CromwellApiServiceSpec extends FlatSpec with CromwellApiService with Scala
       }
   }
 
+  "CromwellApiService" should "return 404 for abort of unknown workflow" in {
+    Post(s"/workflow/${MockWorkflowManagerActor.unknownId}/abort") ~>
+      abortRoute ~>
+      check {
+        assertResult(StatusCodes.NotFound) {
+          status
+        }
+      }
+  }
+
+  it should "return 400 for abort of a malformed workflow id" in {
+    Post(s"/workflow/foobar/abort") ~>
+      abortRoute ~>
+      check {
+        assertResult(StatusCodes.BadRequest) {
+          status
+        }
+      }
+  }
+
+  it should "return 200 for abort of a known workflow id" in {
```

Comment:
I'd like to update this to handle terminal workflows, does 403 seem like the right status?